### ISSUE TITLE
[Bug]: [benchmark] insert 100 million data, concurrent operations, querynode has memory burrs

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#23794]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230427-c82ec0b1 
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.7dev3
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

index:hnsw
case:  test_concurrent_locust_100m_hnsw_ddl_dql_filter_kafka_cluster  

server
```
fouramf-r6xks-8-4777-etcd-0                                       1/1     Running            0               2m47s   10.104.21.109   4am-node24   <none>           <none>
fouramf-r6xks-8-4777-etcd-1                                       1/1     Running            0               2m47s   10.104.23.187   4am-node27   <none>           <none>
fouramf-r6xks-8-4777-etcd-2                                       1/1     Running            0               2m47s   10.104.4.151    4am-node11   <none>           <none>
fouramf-r6xks-8-4777-kafka-0                                      1/1     Running            3 (92s ago)     2m47s   10.104.4.147    4am-node11   <none>           <none>
fouramf-r6xks-8-4777-kafka-1                                      1/1     Running            3 (84s ago)     2m47s   10.104.9.175    4am-node14   <none>           <none>
fouramf-r6xks-8-4777-kafka-2                                      1/1     Running            3 (72s ago)     2m47s   10.104.6.201    4am-node13   <none>           <none>
fouramf-r6xks-8-4777-milvus-datacoord-6858c84f79-fbmk8            1/1     Running            0               2m47s   10.104.9.171    4am-node14   <none>           <none>
fouramf-r6xks-8-4777-milvus-datanode-864486c97-v4hjd              1/1     Running            0               2m47s   10.104.19.156   4am-node28   <none>           <none>
fouramf-r6xks-8-4777-milvus-indexcoord-bf6b6f54b-465d7            1/1     Running            0               2m47s   10.104.9.172    4am-node14   <none>           <none>
fouramf-r6xks-8-4777-milvus-indexnode-55484bd57f-2zxks            1/1     Running            0               2m47s   10.104.20.84    4am-node22   <none>           <none>
fouramf-r6xks-8-4777-milvus-proxy-66b546fd9b-8wl2l                1/1     Running            0               2m47s   10.104.20.83    4am-node22   <none>           <none>
fouramf-r6xks-8-4777-milvus-querycoord-786f8d89fc-55n4w           1/1     Running            0               2m47s   10.104.9.170    4am-node14   <none>           <none>
fouramf-r6xks-8-4777-milvus-querynode-76fd784799-fh5w7            1/1     Running            0               2m47s   10.104.1.231    4am-node10   <none>           <none>
fouramf-r6xks-8-4777-milvus-querynode-76fd784799-rtv2x            1/1     Running            0               2m47s   10.104.9.173    4am-node14   <none>           <none>
fouramf-r6xks-8-4777-milvus-rootcoord-55dd649c7c-jhtmk            1/1     Running            0               2m47s   10.104.1.230    4am-node10   <none>           <none>
fouramf-r6xks-8-4777-minio-0                                      1/1     Running            0               2m47s   10.104.21.110   4am-node24   <none>           <none>
fouramf-r6xks-8-4777-minio-1                                      1/1     Running            0               2m47s   10.104.4.150    4am-node11   <none>           <none>
fouramf-r6xks-8-4777-minio-2                                      1/1     Running            0               2m47s   10.104.23.188   4am-node27   <none>           <none>
fouramf-r6xks-8-4777-minio-3                                      1/1     Running            0               2m47s   10.104.18.165   4am-node25   <none>           <none>
fouramf-r6xks-8-4777-zookeeper-0                                  1/1     Running            0               2m47s   10.104.20.86    4am-node22   <none>           <none>
fouramf-r6xks-8-4777-zookeeper-1                                  1/1     Running            0               2m47s   10.104.23.185   4am-node27   <none>           <none>
fouramf-r6xks-8-4777-zookeeper-2                                  1/1     Running            0               2m47s   10.104.21.112   4am-node24   <none>           <none>
```

querynode memory:

![image](https://user-images.githubusercontent.com/48523564/235058499-a156e4f7-55bf-4b47-b44c-df64b270d258.png)

### Steps To Reproduce

```markdown
1. create a collection or use an existing collection
        2. build index on vector column
        3. insert a certain number of vectors
        4. flush collection
        5. build index on vector column with the same parameters
        6. build index on on scalars column or not
        7. count the total number of rows
        8. load collection
        9. perform concurrent operations (load,query,drop and create collection, search)
        10. clean all collections or not
```

**Comment by yanliang567** [comment]
/assign @jiaoew1991 
is it doing compaction or something?
/unassign

**Comment by elstic** [comment]
> /assign @jiaoew1991 is it doing compaction or something? /unassign

I checked the point in time when the memory node was highlighted and selected two, and their logs were both knowhere in the load index . 

![image](https://user-images.githubusercontent.com/48523564/235079166-efff99de-28fd-46bc-992c-c1633bb77a88.png)
![image](https://user-images.githubusercontent.com/48523564/235079816-405b3bd7-a8ae-4315-9c1c-eccfa0b401fe.png)

**Comment by aoiasd** [comment]
related:  https://github.com/milvus-io/milvus/issues/23454
fixed in:  https://github.com/milvus-io/milvus/pull/23806

**Comment by elstic** [comment]
Verified and fixed, image: 2.2.0-20230506-9fc1ce14


![image](https://user-images.githubusercontent.com/48523564/236719058-967028f5-edfc-47ed-b3c5-06a5f8a3fae4.png)
